examples/votchallenge/main.rs

```diff
@@ -0,0 +1,142 @@
+};
+
+#[derive(Debug)]
+pub enum ServerState {
+    Introduction,
+    Initialization,
+    Reporting {
+        multi_tracker: MultiMosseTracker,
+        first_region: Region,
+    },
+    Termination,
+}
+
+struct MosseTraxServer {
+    state: ServerState,
+}
+impl Default for MosseTraxServer {
+    fn default() -> Self {
+        Self {
+            state: ServerState::Introduction,
+        }
+    }
+}
+impl MosseTraxServer {
+    fn run(mut self) {
+        log::info!("starting run");
+
+        println!("{}", self.make_hello_message());
+
+        for line in stdin().lines() {
+            let line = line.unwrap();
+            log::trace!("handling line: {line:?}");
+            let message: TraxMessageFromClient = line.parse().unwrap();
+            let response = self.process_message(message);
+            println!("{}", response);
+        }
+    }
+
+    fn make_hello_message(&mut self) -> TraxMessageFromServer {
+        TraxMessageFromServer::Hello {
+            version: 1,
+            name: "MosseRust".to_string(),
+            identifier: "mosse-tracker-rust".to_string(),
+            image: ImageType::Path,
+            region: RegionType::Rectangle,
+            channels: vec![ChannelType::Color],
+        }
+    }
+
+    fn process_message(&mut self, message: TraxMessageFromClient) -> TraxMessageFromServer {
+        match message {
+            TraxMessageFromClient::Initialize { image, region } => self.process_init(image, region),
+            TraxMessageFromClient::Frame { images } => self.process_frame(images),
+            // FIXME: return Result from this function, and make the outer loop print "quit" and exit on error?
+            TraxMessageFromClient::Quit => panic!("client sent quit message"),
+        }
+    }
+
+    fn process_init(&mut self, image: Image, region: Region) -> TraxMessageFromServer {
+        let first = image.open().unwrap();
+
+        // initialize a new model
+        let (width, height) = first.to_rgb8().dimensions();
+        // FIXME: use min(region.width, region.height) here?
+        let window_size = 64; //size of the tracking window
+        let psr_thresh = 7.0; // how high the psr must be before prediction is considered succesful.
+        let settings = MosseTrackerSettings {
+            window_size: window_size,
+            width,
+            height,
+            regularization: 0.001,
+            learning_rate: 0.05,
+            psr_threshold: psr_thresh,
+        };
+        let desperation_threshold = 3; // how many frames the tracker should try to re-acquire the target until we consider it failed
+
+        // FIXME: can I get away with a single MosseTracker here?
+        let mut multi_tracker = MultiMosseTracker::new(settings, desperation_threshold);
+
+        let coords = (
+            (region.left + region.width / 2.) as u32,
+            (region.top + region.height / 2.) as u32,
+        );
+        multi_tracker.add_or_replace_target(0, coords, &first.to_luma8());
```

Comment:
I am pretty sure I have my x and y the wrong way around here. Need to try drawing the region on an image and writing it out to disk. Might also be worth trying to cut out the resulting image swatch and write that out to disk?